pkg/controllers/subnetbinding/subnetbinding_controller.go

```diff
@@ -270,15 +326,82 @@ func (r *Reconciler) validateVpcSubnetsBySubnetCR(ctx context.Context, namespace
 		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByChildSubnet(subnets[0])
 		if len(bms) > 0 {
 			dependency := r.SubnetBindingService.GetSubnetConnectionBindingMapCRName(bms[0])
-			return nil, &errorWithRetry{
+			return nil, subnetCR, &errorWithRetry{
 				message: fmt.Sprintf("Target Subnet CR %s is associated by %s", name, dependency),
 				error:   fmt.Errorf("target Subnet %s is already associated by SubnetConnectionBindingMap %s", name, dependency),
 				retry:   true,
 			}
 		}
 	}
+	return subnets, subnetCR, nil
+}
 
-	return subnets, nil
+func (r *Reconciler) getSharedSubnetFromNSX(subnet *v1alpha1.Subnet) (*model.VpcSubnet, *errorWithRetry) {
+	anno := subnet.GetAnnotations()
+	associatedResource, ok := anno[servicecommon.AnnotationAssociatedResource]
+	if !ok {
+		return nil, &errorWithRetry{
+			message: fmt.Sprintf("No associated resource annotation in Shared Subnet %s/%s", subnet.Namespace, subnet.Name),
+			error:   fmt.Errorf("no associated resource annotation in Shared Subnet %s/%s", subnet.Namespace, subnet.Name),
+			retry:   false,
```

Comment:
Is it possible to add the annotation later?

Reply:
No, we assume the annotation is added when the shared Subnet CR is created; and the shared Subnet CR is only created by NSX Operator.